### PR TITLE
[llvm/unittests] Reset the IsSSA property when using finalizeBundle()

### PR DESCRIPTION
**Pull request by ssahasra**

### CODE REVIEW

File: llvm/unittests/MI/LiveIntervalTest.cpp [hunk 1]
```diff
@@ -156,6 +156,7 @@ static void testHandleMoveIntoNewBundle(MachineFunction &MF, LiveIntervals &LIS,
 
   // Build bundle
   finalizeBundle(MBB, I, std::next(ToInstr.getIterator()));
+  MF.getProperties().reset(MachineFunctionProperties::Property::IsSSA);
```
Comment (arsenm) [inline llvm/unittests/MI/LiveIntervalTest.cpp:159]: The test probably shouldn't have had IsSSA in the first place

Comment (ssahasra) [reply llvm/unittests/MI/LiveIntervalTest.cpp:159]: IsSSA is computed by the MIRParser

Comment (arsenm) [reply llvm/unittests/MI/LiveIntervalTest.cpp:159]: By buggy I mean assume SSA by default runs into these sorts of issues. Tests have a definite SSAness which it gets wrong. You could just as well clear this before finalizeBundle

**Review by arsenm — APPROVED** [review]

**Review by arsenm — APPROVED** [review]

File: llvm/unittests/MI/LiveIntervalTest.cpp (hunk 1 above)
Comment (cdevadas) [reply llvm/unittests/MI/LiveIntervalTest.cpp:159]: I guess @yashssh ran into a similar situation once and was trying to fix the default isSSA property assumed by MIRParser. Wasn't that ever landed?

**Pull request merged**